test/unittests/commands/xmlupload/make_rdf_graph/test_make_resource_and_values.py

```diff
@@ -0,0 +1,103 @@
+from dsp_tools.commands.xmlupload.models.intermediary.resource import MigrationMetadata
+from dsp_tools.commands.xmlupload.models.permission import Permissions
+from dsp_tools.commands.xmlupload.models.permission import PermissionValue
+from dsp_tools.models.datetimestamp import DateTimeStamp
+
+ONTO = Namespace("http://0.0.0.0:3333/ontology/9999/onto/v2#")
+RESOURCE_TYPE = URIRef("http://0.0.0.0:3333/ontology/9999/onto/v2#TestResource")
+LABEL = Literal("Resource Label", datatype=XSD.string)
+PROJECT_IRI = URIRef("http://rdfh.ch/9999/project")
+
+
+@pytest.fixture
+def migration_metadata() -> MigrationMetadata:
+    return MigrationMetadata(
+        "http://rdfh.ch/4123/DiAmYQzQSzC7cdTo6OJMYA", DateTimeStamp("1999-12-31T23:59:59.9999999+01:00")
+    )
+
+
+def test_make_resource_mandatory_only() -> None:
+    res = IntermediaryResource(
+        res_id="resource_id",
+        type_iri="http://0.0.0.0:3333/ontology/9999/onto/v2#TestResource",
+        label="Resource Label",
+        permissions=None,
+        values=[],
+        file_value=None,
+        iiif_uri=None,
+        migration_metadata=None,
+    )
+    res_bn = BNode()
+    res_graph = _make_resource(res, res_bn, PROJECT_IRI)
+    assert len(res_graph) == 3
+    assert next(res_graph.objects(res_bn, RDF.type)) == RESOURCE_TYPE
+    assert next(res_graph.objects(res_bn, RDFS.label)) == LABEL
+    assert next(res_graph.objects(res_bn, KNORA_API.attachedToProject)) == PROJECT_IRI
+
+
+def test_make_resource_permissions() -> None:
+    res = IntermediaryResource(
+        res_id="resource_id",
+        type_iri="http://0.0.0.0:3333/ontology/9999/onto/v2#TestResource",
+        label="Resource Label",
+        permissions=Permissions({PermissionValue.CR: ["knora-admin:ProjectAdmin"]}),
+        values=[],
+        file_value=None,
+        iiif_uri=None,
+        migration_metadata=None,
+    )
+    res_bn = BNode()
+    res_graph = _make_resource(res, res_bn, PROJECT_IRI)
+    assert len(res_graph) == 4
+    assert next(res_graph.objects(res_bn, RDF.type)) == RESOURCE_TYPE
+    assert next(res_graph.objects(res_bn, RDFS.label)) == LABEL
+    assert next(res_graph.objects(res_bn, KNORA_API.attachedToProject)) == PROJECT_IRI
+    permissions = next(res_graph.objects(res_bn, KNORA_API.hasPermissions))
+    assert permissions == Literal("CR knora-admin:ProjectAdmin", datatype=XSD.string)
+
+
+def test_make_resource_migration_metadata(migration_metadata: MigrationMetadata) -> None:
+    res = IntermediaryResource(
+        res_id="resource_id",
+        type_iri="http://0.0.0.0:3333/ontology/9999/onto/v2#TestResource",
+        label="Resource Label",
+        permissions=None,
+        values=[],
+        file_value=None,
+        iiif_uri=None,
+        migration_metadata=migration_metadata,
+    )
+    res_iri = URIRef("http://rdfh.ch/4123/DiAmYQzQSzC7cdTo6OJMYA")
+    res_graph = _make_resource(res, res_iri, PROJECT_IRI)
+    assert len(res_graph) == 3
```

Comment:
hmm, but the metadata should also be contained in the graph. Why have they vanished?

Reply:
They have not vanished, but they are not added to the graph in this function. I will remove it, I think when I wrote the test it made more sense.